### PR TITLE
Refs #32508 -- Raised TypeError/ValueError instead of using "assert" in encode() methods of some password hashers.

### PR DESCRIPTION
**Pull request by felixxm**

### CODE REVIEW

File: django/contrib/auth/hashers.py [hunk 1]
```diff
@@ -209,6 +209,12 @@ def verify(self, password, encoded):
         """Check if the given password is correct."""
         raise NotImplementedError('subclasses of BasePasswordHasher must provide a verify() method')
 
+    def _check_encode_args(self, password, salt):
+        if password is None:
+            raise ValueError('password must be provided.')
```
Comment (cjerdonek) [inline django/contrib/auth/hashers.py:214]: Would `TypeError` be more appropriate?

**Review by cjerdonek — COMMENTED** [review]
Thanks, @felixxm.

File: django/contrib/auth/hashers.py (hunk 1 above)
Comment (felixxm) [reply django/contrib/auth/hashers.py]: Agreed.

**Comment by pauloxnet** [comment]
Thanks, @felixxm

**Review by pauloxnet — APPROVED** [review]

**Review by hramezani — APPROVED** [review]
LGTM!

**Review by cjerdonek — APPROVED** [review]

**Comment by felixxm** [comment]
Thanks y'all for checking :+1:

**Pull request merged**